lgn-provers/src/params/mod.rs

```diff
@@ -136,4 +157,34 @@ impl ParamsLoader {
         info!("Stored params to local storage");
         Ok(())
     }
+    fn verify_checksum(file: &File, expected_checksum: &str) -> anyhow::Result<bool> {
+        let mut reader = std::io::BufReader::new(file);
+        let mut hasher = Sha256::new();
+        let mut buffer = Vec::new();
+        reader
+            .read_to_end(&mut buffer)
+            .context("Failed to read file for checksum verification")?;
+        hasher.update(&buffer);
+        let actual_checksum = format!("{:x}", hasher.finalize());
+        if actual_checksum != expected_checksum {
+            info!(
+                "Checksum mismatch: expected {}, but got {}",
+                expected_checksum, actual_checksum
+            );
+        }
+        Ok(actual_checksum == expected_checksum)
```

Comment:
Mhhh can we maybe use standard tools for this ? it's gonna be easier to port accross platforms and easier to write scripts as well in bash. Like `sha256sum` or `shasum` i think. Also probably might be a bit faster.

Reply:
But in this case, we would have to download + save before checking checksum.
Which is fine for me btw.

Another solution would be to [use blake3](https://docs.rs/blake3/latest/blake3/struct.Hasher.html#method.update_rayon) which supports reading from mmap and even do stuff in parallel with rayon and blake3 is known to be much faster, even more for large files.